src/formatDate.js

```diff
@@ -50,7 +50,66 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {
+    separator: fromFormat[fromFormat.length - 1],
+  };
+
+  const separateDate = date.split(dateParts.separator);
+
+  for (const partFrom of fromFormat) {
+    switch (partFrom) {
+      case 'YY':
+        dateParts.year = separateDate[fromFormat.indexOf(partFrom)];
+        break;
+
+      case 'YYYY':
+        dateParts.year = separateDate[fromFormat.indexOf(partFrom)];
+        break;
+
+      case 'MM':
+        dateParts.month = separateDate[fromFormat.indexOf(partFrom)];
+        break;
+
+      case 'DD':
+        dateParts.day = separateDate[fromFormat.indexOf(partFrom)];
+        break;
+    }
+  }
+
+  for (const partTo of toFormat) {
+    switch (partTo) {
+      case 'YY':
+        if (partTo.length === dateParts.year.length) {
+          toFormat[toFormat.indexOf(partTo)] = dateParts.year;
+          break;
+        } else {
+          toFormat[toFormat.indexOf(partTo)]
+          = dateParts.year.slice(2, dateParts.year.length);
```

Comment:
if you write slice(2) only it will work still. Because it starts from 2 and goes till the end anyway.